### PR TITLE
Android tv-casting-app: refactored Initialization

### PR DESCRIPTION
**Pull request by sharadb-amazon**
The TvCastingApp.java and TvCastingApp-JNI.h/cpp in the Android tv-casting-app have grown in size and should be broken down. This PR is the first step towards doing that. It introduces a simplified version of the Initialization code and corresponding APIs.

\### Change summary  
1\. Added an InitializationExample.java under tv-casting-app/android. It has a initAndStart menthod which is called by the MainActivity, if and only if GlobalCastingConstants.ChipCastingSimplified is set to true (false by default). While these APIs are being refactored, we will keep this flag off (can be turned to true for local testing)   
2\. Added a CastingApp.java class where we've added the Initialize(), Start() and Stop() APIs. There are some supporting classes in the "support" namespace and there is a CastingApp-JNI.h/cpp that carries native implementation of a couple of private methods.

Some items are marked TODO - they will be addressed in following PRs, when the dependent workflows (commissioning, CASE, etc) are refactored.

\### Testing  
Tested by building (with the GlobalCastingConstants.ChipCastingSimplified=true flag set) and running the Android tv-casting-app on Mac to see that it initializes and starts the Matter server.

### CODE REVIEW

File: examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/core/CastingApp.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  public MatterError stop() {
+    Log.i(TAG, "CastingApp.stop called");
+    if (mState != CastingAppState.RUNNING) {
+      return MatterError.CHIP_ERROR_INCORRECT_STATE;
+    }
+
+    boolean serverStopped = chipAppServer.stopApp();
+    if (!serverStopped) {
+      Log.e(TAG, "CastingApp.stop failed to stop Matter server");
+      return MatterError.CHIP_ERROR_INCORRECT_STATE;
```
Comment (chrisdecenzo) [inline examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/core/CastingApp.java:142]: is this the most accurate error for this scenario?

File: examples/tv-casting-app/android/App/app/src/main/jni/cpp/core/CastingApp-JNI.cpp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    return support::createJMatterError(CHIP_NO_ERROR);
+}
+
+JNI_METHOD(jobject, finishStartup)(JNIEnv *, jobject)
+{
+    chip::DeviceLayer::StackLock lock;
+    ChipLogProgress(AppServer, "JNI_METHOD CastingAppJNI.finishStartup called");
+    auto & server = chip::Server::GetInstance();
+
+    // TODO: Set AppDelegate
```
Comment (chrisdecenzo) [inline examples/tv-casting-app/android/App/app/src/main/jni/cpp/core/CastingApp-JNI.cpp:84]: Are these TODO's optional? Just wondering if this means there is missing functionality right now.

**Review by chrisdecenzo — APPROVED** [review]

**Review by hawk248 — APPROVED** [review]

File: examples/tv-casting-app/android/App/app/src/main/jni/cpp/core/CastingApp-JNI.cpp (hunk 2 above)
Comment (sharadb-amazon) [reply examples/tv-casting-app/android/App/app/src/main/jni/cpp/core/CastingApp-JNI.cpp:84]: Not optional. I'll be refactoring some of more the old code into new classes in following PRs, and will then complete these TODOs.

File: examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/core/CastingApp.java (hunk 1 above)
Comment (sharadb-amazon) [reply examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/core/CastingApp.java:142]: Given that the underlying ChipAppServer in the core SDK that we are using here, is returning a hardcoded boolean (JNI\_TRUE), CHIP\_ERROR\_INCORRECT\_STATE seemed appropriate to return.

**Pull request merged**